pkg/ui/react-app/src/components/ListTree.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { InputProps, Collapse, ListGroupItem, ListGroup } from 'reactstrap';
+
+export interface NodeTree {
+  name: string;
+  children?: NodeTree[];
+}
+
+interface NodeProps extends InputProps {
+  node: NodeTree | null;
+}
+
+const ListTree: React.FC<NodeProps> = ({ id, node }) => {
+  type mapping = {
+    [key: string]: boolean;
+  };
+
+  const [state, setState] = useState<mapping>({});
```

Comment:
Let's make naming a bit more descriptive here!
```suggestion
  const [mapping, setMapping] = useState<mapping>({});
```